beacon_node/http_api/tests/tests.rs

```diff
@@ -3507,31 +3507,58 @@ impl ApiTester {
                 sk.sign(message).into()
             };
 
-            let block = self
+            let (response, metadata) = self
                 .client
-                .get_validator_blinded_blocks::<E>(slot, &randao_reveal, None)
+                .get_validator_blocks_v3::<E>(slot, &randao_reveal, None, None)
                 .await
-                .unwrap()
-                .into_data();
+                .unwrap();
 
-            let signed_block = block.sign(&sk, &fork, genesis_validators_root, &self.chain.spec);
+            match response.data {
+                ProduceBlockV3Response::Blinded(blinded_block) => {
+                    assert!(metadata.execution_payload_blinded);
+                    assert_eq!(
+                        metadata.consensus_version,
+                        blinded_block.to_ref().fork_name(&self.chain.spec).unwrap()
+                    );
+                    let signed_blinded_block =
+                        blinded_block.sign(&sk, &fork, genesis_validators_root, &self.chain.spec);
 
-            self.client
-                .post_beacon_blinded_blocks(&signed_block)
-                .await
-                .unwrap();
+                    self.client
+                        .post_beacon_blinded_blocks(&signed_blinded_block)
+                        .await
+                        .unwrap();
 
-            let head_block = self
-                .client
-                .get_beacon_blocks(CoreBlockId::Head)
-                .await
-                .unwrap()
-                .unwrap()
-                .into_data();
+                    let head_block = self.chain.head_beacon_block().clone_as_blinded();
+                    assert_eq!(head_block, signed_blinded_block);
 
-            assert_eq!(head_block.clone_as_blinded(), signed_block);
+                    self.chain.slot_clock.set_slot(slot.as_u64() + 1);
+                }
+                ProduceBlockV3Response::Full(block_contents) => {
+                    assert!(!metadata.execution_payload_blinded);
```

Comment:
This test is meant to be checking blinded production, so I think it should error in the `Full` case